libraries/kotlin.test/common/src/main/kotlin/kotlin/test/Assertions.kt

```diff
@@ -72,6 +72,21 @@ fun <@OnlyInputTypes T> assertNotSame(illegal: T, actual: T, message: String? =
     asserter.assertNotSame(message, illegal, actual)
 }
 
+/** Asserts that [value] is [T], with an optional [message]. */
+@SinceKotlin("1.4")
+@InlineOnly
+inline fun <reified T: Any> assertIs(value: Any, message: String? = null) {
+    contract { returns() implies (value is T) }
+    asserter.assertTrue({ messagePrefix(message) + "Expected value to be of type <${T::class}>, actual <${value::class}>." }, value is T)
```

Comment:
`typeOf<T>()` provides more precise information about the type, especially its nullability.